Sources/Commands/SwiftRunTool.swift

```diff
@@ -128,9 +163,18 @@ public class SwiftRunTool: SwiftTool<RunToolOptions> {
     }
     
     /// Executes the executable at the specified path.
-    private func run(_ excutablePath: AbsolutePath) throws {
-        let pathRelativeToWorkingDirectory = excutablePath.relative(to: originalWorkingDirectory)
-        try exec(path: excutablePath.asString, args: [pathRelativeToWorkingDirectory.asString] + options.arguments)
+    private func run(_ excutablePath: AbsolutePath, arguments: [String]) throws {
+        let pathRelativeToWorkingDirectory = excutablePath.relative(to: currentWorkingDirectory)
+        try exec(path: excutablePath.asString, args: [pathRelativeToWorkingDirectory.asString] + arguments)
+    }
+
+    /// Determines if a path points to a valid swift file.
+    private func isValidSwiftFilePath(_ path: String) -> Bool {
+        guard path.hasSuffix(".swift") else { return false }
```

Comment:
Yes. I think it does some amount of checking but asserts in case of errors. Here we want something which throws because `path` is an user input.